### PR TITLE
Make enhancements

### PR DESCRIPTION
**Pull request by elevran**
**What this PR does / why we need it**:
enhancements to Makefile

**Which issue this PR fixes**: NONE

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,17 +56,80 @@ checkvars:
...
 
-fmt:
-	bin/fmt.sh
+#-----------------------------------------------------------------------------
+# Target: precommit
+#-----------------------------------------------------------------------------
+.PHONY: clean
+.PHONY: clean.bazel clean.go
+
+clean: clean.bazel
```
Comment (rshriram) [inline Makefile]: shouldn't this be clean.go?

Comment (elevran) [reply Makefile]: Keeping the clean target backward compatible. Can always run `make clean.go` directly

**Pull request merged**